### PR TITLE
rye -> uv +  Python 3.11 -> 3.10

### LINKED ISSUE #977: Migrate from Rye to UV

### LINKED ISSUE #1061: Downgrade python version to 3.10
This seems to be a more widely used version of python, especially for notebooks

### PR DESCRIPTION
**Pull request by AdamGS**
Closes #1061 
Closes #977

Moves us from rye to uv + lowers the target python version.

### CODE REVIEW

File: uv.lock [hunk 1]
```diff
@@ -0,0 +1,1414 @@
+version = 1
```
Comment (AdamGS) [inline uv.lock:1]: do we actually need this file?

**Review by robert3005 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -43,13 +43,13 @@ jobs:
 
       - name: Pytest - PyVortex
         run: |
-          rye run maturin develop --skip-install
-          rye run pytest --benchmark-disable test/
+          uv run maturin develop --skip-install
```
Comment (gatesn) [inline .github/workflows/ci.yml]: I don't think we need Maturin develop?`uv run` should do it automatically no?

Comment (AdamGS) [reply .github/workflows/ci.yml]: yeah definetly looks like it, removing

File: uv.lock (hunk 1 above)
Comment (AdamGS) [reply uv.lock:1]: keeping for now seems fine

**Pull request merged**